### PR TITLE
javadoc for AZ files and DictionaryDatabase

### PR DESCRIPTION
**Pull request by arangob**
+renamed my A-Z files to more appropriate/fitting names
+added javadoc for A-Z files and DictionaryDatabase

### CODE REVIEW

File: app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java [hunk 1]
```diff
@@ -8,28 +8,52 @@
...
+         */
+        public ViewHolder(CardView cardView) {
+            super(cardView);
+            this.cardView = cardView;
         }
     }
 
-    public CategoryAdapter(String[] categories){
+    /**
+     * Class constructor specifying the list of categories to used in a CardView.
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java:44]: Class constructor specifying the list of categories to use in a CardView.

File: app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java [hunk 2]
```diff
@@ -8,28 +8,52 @@
 import android.view.ViewGroup;
 import android.widget.TextView;
 
-class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.ViewHolder> {
-    private final static String TAG = "CategoryAdapter";
+/**
+ * AZCategoryAdapter implements the adapter for the RecyclerView in AZCategoryFragment.
+ */
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java:13]: I noticed Ellen added this line to her class files:
* @author Ellen Spertus (ellen.spertus@gmail.com)

Comment (RobertoAF) [reply app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java:13]: * @author Barbara Arango (barango@mills.edu)

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java [hunk 3]
```diff
@@ -18,36 +16,46 @@
 import android.view.ViewGroup;
 import android.widget.Toast;
 
-public class AZCategoryMaterialList extends Fragment {
-    private static final String TAG = "AZCategoryMaterialList";
-    private AZTermList azTermList;
+/**
+ * AZCategoryFragment implements the view for categories within the dictionary database. Categories are
+ * displayed in a recycler view containing card view's.
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:21]: * @author Barbara Arango (barango@mills.edu)

Comment (RobertoAF) [reply app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:21]: * displayed in a recycler view containing card views[card's views? or card views?].

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java [hunk 4]
```diff
@@ -62,25 +70,28 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             toast = Toast.makeText(this.getContext(), "Database Unavailable", Toast.LENGTH_LONG);
             toast.show();
         }
+    }
 
-
-        CategoryAdapter categoryAdapter = new CategoryAdapter(categories);
-        recyclerView.setAdapter(categoryAdapter);
+    /**
+     * Creates the RecyclerView adapter for displaying category CardView's.
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:76]: Creates the RecyclerView adapter for displaying category CardView's[CardViews?].

File: app/src/main/java/edu/mills/cs115/AZTermListFragment.java [hunk 5]
```diff
@@ -94,8 +103,13 @@ public void onListItemClick(ListView l, View v, int position, long id) {
         getActivity().startActivity(intent);
     }
 
-    public void setCategory(int category) {
-        switch (category) {
+    /**
+     * Sets the string representation of a category by the position in which was selected.
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZTermListFragment.java:107]: * Sets the string representation of a category by the position in which [it?] was selected.

File: app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java [hunk 6]
```diff
@@ -12,12 +12,13 @@
 import java.io.InputStream;
 import java.io.InputStreamReader;
 
-
+/**
+ * DictionaryDatabaseHelper implements the database for dictionary table .
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java:16]: * DictionaryDatabaseHelper implements the database for dictionary table.

File: app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java [hunk 7]
```diff
@@ -29,11 +30,15 @@
     static final String DEFINITION_COL = "definition";
     static final String CATEGORY_COL = "category";
     static final String FAVORITES_COL = "favorites";
-    static final String NOTES_COL = "notes";
 
+    /**
+     * Default constructor
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java:35]: * Default constructor.

File: app/src/main/java/edu/mills/cs115/AZTermListFragment.java [hunk 8]
```diff
@@ -14,8 +14,12 @@
 import android.widget.ListView;
 import android.widget.Toast;
 
-public class AZTermList extends ListFragment {
-    private static final String TAG = "AZTermList";
+/**
+ * AZTermListFragment implements a ListFragment representation of all the terms categorized under a
+ * selected category.
+ */
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZTermListFragment.java:20]: @author Barbara Arango (barango@mills.edu)

File: app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java [hunk 9]
```diff
@@ -12,12 +12,13 @@
 import java.io.InputStream;
 import java.io.InputStreamReader;
 
-
+/**
+ * DictionaryDatabaseHelper implements the database for dictionary table .
+ */
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java:17]: @author Barbara Arango (barango@mills.edu)

File: app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java [hunk 10]
```diff
@@ -8,28 +8,52 @@
 import android.view.ViewGroup;
 import android.widget.TextView;
 
-class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.ViewHolder> {
-    private final static String TAG = "CategoryAdapter";
+/**
+ * AZCategoryAdapter implements the adapter for the RecyclerView in AZCategoryFragment.
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java:12]: Do you know if she wants us to provide links to other Activity files? I noticed she does this in her Javadoc comments. Ex:
* AZCategoryAdapter implements the adapter for the RecyclerView in {@link AZCategoryFragment}

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java [hunk 11]
```diff
@@ -1,11 +1,9 @@
 package edu.mills.cs115;
 
-import android.annotation.TargetApi;
 import android.database.Cursor;
 import android.database.sqlite.SQLiteDatabase;
 import android.database.sqlite.SQLiteException;
 import android.database.sqlite.SQLiteOpenHelper;
-import android.os.Build;
 import android.os.Bundle;
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZCategoryFragment.java]: Will you leave these lines blank in order to organize/categorize the import statements?

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java [hunk 12]
```diff
@@ -1,11 +1,9 @@
 package edu.mills.cs115;
```
Comment (RobertoAF) [inline app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:2]: Will you leave these lines blank in order to organize/categorize the import statements?

File: app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java (hunk 1 above)
Comment (RobertoAF) [reply app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java:44]: +check spelling

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java (hunk 12 above)
Comment (RobertoAF) [reply app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:2]: + check empty lines

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java (hunk 3 above)
Comment (RobertoAF) [reply app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:21]: +check use of apostrophe

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java (hunk 4 above)
Comment (RobertoAF) [reply app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:76]: +check use of apostrophe

File: app/src/main/java/edu/mills/cs115/AZTermListFragment.java (hunk 5 above)
Comment (arangob) [reply app/src/main/java/edu/mills/cs115/AZTermListFragment.java:107]: this is getting embarrassing

File: app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java (hunk 10 above)
Comment (arangob) [reply app/src/main/java/edu/mills/cs115/AZCategoryAdapter.java:12]: hmmm i'll add it

File: app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java (hunk 6 above)
Comment (RobertoAF) [reply app/src/main/java/edu/mills/cs115/DictionaryDatabaseHelper.java:16]: +check extra space after table

**Review by RobertoAF — CHANGES_REQUESTED** [review]
So, I added some comments. One thing that I noticed about Ellen's javadoc is that she adds {@link} and {@code} statements. She also adds examples in her comments. Take a look at her javadoc comments as reference just in case you also want to add those.

File: app/src/main/java/edu/mills/cs115/AZCategoryFragment.java (hunk 12 above)
Comment (arangob) [reply app/src/main/java/edu/mills/cs115/AZCategoryFragment.java:2]: yes

**Comment by arangob** [comment]
thank you.

**Comment by RobertoAF** [comment]
One step closer....

**Pull request closed without merging**